lib/src/realm_class.dart

```diff
@@ -113,12 +115,53 @@ class Realm implements Finalizable {
   /// Opens a `Realm` using a [Configuration] object.
   Realm(Configuration config) : this._(config);
 
-  Realm._(this.config, [RealmHandle? handle, this._isInMigration = false]) : _handle = handle ?? _openRealm(config) {
+  Realm._(this.config, [RealmHandle? handle, this._isInMigration = false]) : _handle = handle ?? _openRealmSync(config) {
     _populateMetadata();
     isFrozen = realmCore.isFrozen(this);
   }
 
-  static RealmHandle _openRealm(Configuration config) {
+  /// A method for asynchronously opening a [Realm].
+  ///
+  /// When the configuration is [FlexibleSyncConfiguration], the realm will be downloaded and fully
+  /// synchronized with the server prior to the completion of the returned [Future].
+  /// This method could be called also for opening a local [Realm] with [LocalConfiguration].
+  ///
+  /// * `config`- a configuration object that describes the realm.
+  /// * `cancellationToken` - an optional [CancellationToken] used to cancel the operation.
+  /// * `onProgressCallback` - a callback for receiving download progress notifications for synced [Realm]s.
+  ///
+  /// Returns `Future<Realm>` that completes with the [Realm] once the remote [Realm] is fully synchronized or with a [CancelledException] if operation is canceled.
+  /// When the configuration is [LocalConfiguration] this completes right after the local [Realm] is opened or if the operation is canceled in advance.
+  /// Since opening a local Realm is a synchronous operation, there is no benefit of using Realm.open over the constructor.
+  static Future<Realm> open(Configuration config, {CancellationToken? cancellationToken, ProgressCallback? onProgressCallback}) async {
+    if (cancellationToken?.isCancelled ?? false) {
+	      return Future<Realm>.error(cancellationToken!.exception);
+	    }
```

Comment:
One last thing :-) ..
```suggestion
    if (cancellationToken != null && cancellationToken.isCancelled) {
      throw cancellationToken.exception;
    }
```
since we are in a `async` method, we can just throw (a bit shorter). Also by testing for `!= null` the compiler can infer we don't need the bangs (`!`) later

Reply:
We don't throw because the exceprion occurs when Realm.open is called instead of after await the future returned by Realo.open.

Reply:
The two pieces of code does the exact same thing? .. the methods is marked `async`

Reply:
Right, my previous test was wrong.

Reply:
It is applied.